roles/postgres/tasks/upgrade_postgres.yml

```diff
@@ -123,12 +122,34 @@
   set_fact:
     postgres_pod_name: "{{ postgres_pod['resources'][0]['metadata']['name'] }}"
 
+- name: Get the name of the service for the old postgres pod
+  k8s_info:
+    kind: Service
+    namespace: "{{ ansible_operator_meta.namespace }}"
+    label_selectors:
+      - "app.kubernetes.io/component=database"
+      - "app.kubernetes.io/instance=postgres-{{ ansible_operator_meta.name }}"
+      - "app.kubernetes.io/version={{ supported_pg_version }}"
+      - "app.kubernetes.io/managed-by={{ deployment_type }}-operator"
+  register: old_postgres_svc
+
+# TODO: The galaxy-operator labels are different. potential issue here.
+- name: Get the name of the service for the old postgres pod
+  k8s_info:
+    kind: Service
+    namespace: "{{ ansible_operator_meta.namespace }}"
+    label_selectors:
+      - "app.kubernetes.io/component=database"
+      - "app.kubernetes.io/instance={{ old_postgres_pod.metadata.labels['app.kubernetes.io/instance'] }}"
+      - "app.kubernetes.io/version={{ old_postgres_pod.metadata.labels['app.kubernetes.io/version'] }}"  # unique to the galaxy-operator
+      - "app.kubernetes.io/managed-by={{ deployment_type }}-operator"
+  register: old_postgres_svc
```

Comment:
so we're trying to get information from the old service but we register to the same variable so anything registered in the initial task is erased.
Is that expected ? is this some preference order ?
Also the first task is using `"app.kubernetes.io/version={{ supported_pg_version }}"` but at this point we already have the `supported_pg_version` variable sets to 15 right ? in this case this isn't what we want, we want the old one from either postgresql 12 or 13 ?

Reply:
This first task should just be removed entirely, it was an intermediate task that never got removed. Good find.  I'll remove this now.